philips_hue.py

```diff
@@ -0,0 +1,128 @@
+        username: Authorized user string to make API calls.
+    """
+    def __init__(self, bridge_ip_address, username):
+        self._bridge_ip_address = bridge_ip_address
+        self._username = username
+
+
+    @property
+    def bridge_ip_address(self):
+        return self._bridge_ip_address
+
+
+    @property
+    def username(self):
+        return self._username
+
+
+    def set_color(self, light_id, hue):
+        """Sets the color of the light to a specified hue value.
+
+        Args:
+            light_id:  The id to pass to the Philips Hue API to
+                specify the light to set a color for.
+            hue: Hue of the light. Takes values from 0 to 65535.
+                Programming 0 and 65535 would mean that the light will resemble
+                the color red, 21845 for green and 43690 for blue.
+
+        Returns:
+            HTTP Response from the Philips Hue API.
+        """
+        return requests.put(url=f'http://{self._bridge_ip_address}/api/{self._username}/lights/{light_id}/state',
+                            data=json.dumps({"on": True, "hue": hue}))
+
+
+# TODO(https://github.com/googleinterns/cloud-monitoring-notification-delivery-integration-sample-code/issues/10):
+# Currently specific to Philips Hue, but will be generalized to trigger
+# whatever notification system the client chooses.
+def trigger_light_from_monitoring_notification(client, notification, light_id):
+    """Changes the color of a Philips Hue light based on a monitoring notification
+    and returns the response.
+
+    The color of the Philips Hue light is set through the client,
+    which makes an HTTP PUT request to the Philips Hue API.
+    Sets the color of the light to red if the incident is open and green if the incident is closed.
+
+    Args:
+        client: The PhilipsHueClient object to trigger a response from.
+        notification: A dictionary containing the notification data.
+        light_id: The id of the light to set the color for.
+
+    Returns:
+        A response string indicating whether the light was triggered successfully,
+        as well as the incident state from the notification.
+        An example response is as follows:
+        {'success': 'incident condition state is open'}
+
+    Raises:
+        UnknownConditionStateError: If the incident state is not open or closed.
+        BadAPIRequestError: If there was an error in calling the Philips Hue API.
+    """
+    try:
+        condition_state = notification["incident"]["condition"]["state"]
+    except KeyError:
+        raise NotificationParseError("Notification is missing required dict key")
+
+    if condition_state == "open":
+        response = client.set_color(light_id, 0)  # set to red
+        if response.status_code == 200:
+            return "{'success': 'incident condition state is open'}"
+        else:
+            raise BadAPIRequestError(response.text)
```

Comment:
Two things:

1. Why not move this into the set_color() function? This is essentially a conversion process from HTTP errors to Python errors. Converting between Python and HTTP sounds like a good job for PhilipsHueClient(). It would also move this logic into one centralized place, instead of duplicates scattered around the calling code.
2. The conditional statement can also be simplified to:
```
if response.status_code != 200:
     raise()
return response